OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -159,31 +161,27 @@ protected virtual void Tick(Actor self)
 			dirty = false;
 		}
 
+		void INotifyBecomingIdle.OnBecomingIdle(Actor self)
+		{
+			if (state != AnimationState.Waiting)
+				PlayStandAnimation(self);
+		}
+
 		void INotifyIdle.TickIdle(Actor self)
 		{
-			if (state != AnimationState.Idle && state != AnimationState.IdleAnimating && state != AnimationState.Attacking)
+			if (!AllowIdleAnimation(self))
```

Comment:
Would it make sense here to return early if we don't have an idle animation, i.e. `if (!hasIdleSequence || !AllowIdleAnimation(self))`?

Reply:
`AllowIdleAnimation` already checks `hasIdleSequence`.

Reply:
My bad.